### DISCUSSION TITLE
Add a new group type to palette

### DISCUSSION THREAD

**Discussion by Adem-Kri** [eclipse-glsp/glsp#1132] [Q&A]
Hello, 
I'm wondering how to add a new group type to the palette. (short question)

thanks in advance

![image](https://github.com/eclipse-glsp/glsp/assets/116633370/c64e559f-f77c-4fb0-809b-e465afc8c769)

**Comment by planger** [comment]
By default, the GLSP server module binds the [`DefaultToolPaletteItemProvider`](https://github.com/eclipse-glsp/glsp-server-node/blob/main/packages/server/src/common/features/contextactions/tool-palette-item-provider.ts#L62), which obtains the registered operation handlers and derives the items it shows from that. However, you can easily bind your custom `ToolPaletteItemProvider` by overwriting [bindToolPaletteItemProvider()](https://github.com/eclipse-glsp/glsp-server-node/blob/main/packages/server/src/common/di/diagram-module.ts#L403C15-L403C42) in your server module to specify items, categories, icons, etc. yourself any way you want.

See also the answer to a similar question: https://github.com/eclipse-glsp/glsp/discussions/550

**Comment by rsoika** [comment]
You can take a look into our implementation in [Open-BPMN](https://www.open-bpmn.org/). We also use categories and a custom sorting too. And yes, this was not easy to figure out ;-)
Maybe this may help you: https://github.com/imixs/open-bpmn/blob/master/open-bpmn.glsp-server/src/main/java/org/openbpmn/glsp/provider/BPMNToolPaletteItemProvider.java

![image](https://github.com/eclipse-glsp/glsp/assets/1339124/45bff258-40d0-4755-bfe6-5afdd1e075f1)
